hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/TestCleaner.java

```diff
@@ -228,32 +212,63 @@ public void testInsertAndCleanFailedWritesByVersions() throws Exception {
   }
 
   /**
-   * Test Clean-By-Versions using prepped versions of insert/upsert API.
+   * Test Helper for cleaning failed writes by versions logic from HoodieWriteClient API perspective.
+   *
+   * @param insertFn     Insert API to be tested
+   * @param isPreppedAPI Flag to indicate if a prepped-version is used. If true, a wrapper function will be used during
+   *                     record generation to also tag the regards (de-dupe is implicit as we use unique record-gen APIs)
+   * @throws Exception in case of errors
    */
-  @Test
-  public void testInsertPreppedAndCleanByVersions() throws Exception {
-    testInsertAndCleanByVersions(SparkRDDWriteClient::insertPreppedRecords, SparkRDDWriteClient::upsertPreppedRecords,
-        true);
-  }
+  private void testInsertAndCleanFailedWritesByVersions(
```

Comment:
this moved up to be below `public void testInsertAndCleanFailedWritesByVersions()`